options.go

```diff
@@ -210,7 +225,7 @@ func (opt *poptions) storagePathForPartition(topic string, partitionID int32) st
 }
 
 func (opt *poptions) defaultStorageBuilder(topic string, partition int32, codec Codec, reg metrics.Registry) (storage.Storage, error) {
-	return storage.New(opt.storagePathForPartition(topic, partition), codec, reg, opt.storageSnapshotInterval)
+	return storage.New(logger.Default(), opt.storagePathForPartition(topic, partition), codec, reg, opt.storageSnapshotInterval)
```

Comment:
this should use the option's logger instead of the Default one, otherwise setting a logger does not work for the storage because it always uses the default one.